### PR TITLE
Copy conda-forge.yml from the source recipe directory to the feedstock

### PR DESCRIPTION
**Pull request by mariusvniekerk**
This allows us to more easily automatically enroll feedstocks in the various bots.

Checklist
* [x] Added a ``news`` entry

### CODE REVIEW

File: conda_smithy/cli.py [hunk 1]
```diff
@@ -52,6 +53,23 @@ def generate_feedstock_content(target_directory, source_recipe_dir):
...
+        os.unlink(os.path.join(target_recipe_dir, "conda-forge.yml"))
+        try:
+            with open(forge_yml_recipe, 'r') as fp:
+                _cfg = yaml.load(fp.read())
+        except:
+            _cfg = {}
+
+        with open(forge_yml, 'r') as fp:
+            _cfg_feedstock = yaml.load(fp.read())
+            _cfg_feedstock.update(_cfg)
```
Comment (beckermr) [inline conda_smithy/cli.py:69]: I think we want a recursive update here to allow us to replaces bits multiply nested keys.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: conda_smithy/cli.py (hunk 1 above)
Comment (mariusvniekerk) [reply conda_smithy/cli.py:69]: yeah, i guess, though atm in almost all cases this is {}

File: conda_smithy/cli.py [hunk 2]
```diff
@@ -52,6 +53,23 @@ def generate_feedstock_content(target_directory, source_recipe_dir):
...
+            with open(forge_yml_recipe, "r") as fp:
+                _cfg = yaml.load(fp.read())
+        except:
+            _cfg = {}
+
+        with open(forge_yml, "r") as fp:
+            _cfg_feedstock = yaml.load(fp.read())
+            merge_dict(_cfg, _cfg_feedstock)
+        with open(forge_yml, "w") as fp:
+            yaml.dump(_cfg_feedstock, fp)
```
Comment (isuruf) [inline conda_smithy/cli.py:71]: I think you need to git add the file afterwards. You can do `yaml.dumps` and use `feedstock_io.write_file`

Comment (mariusvniekerk) [reply conda_smithy/cli.py:71]: I think the commit on https://github.com/conda-forge/conda-smithy/blob/589689ed9af2cf1fbaaf4e9ae39f0b042e291382/conda_smithy/cli.py#L139 takes care of that

Comment (isuruf) [reply conda_smithy/cli.py:71]: I mean the `git add`, not `git commit`.

Comment (mariusvniekerk) [reply conda_smithy/cli.py:71]: we don't actually explicitly call git add anywhere in that phase

Comment (beckermr) [reply conda_smithy/cli.py:71]: `feedstock_io.write_file` does the `git add`

Comment (beckermr) [reply conda_smithy/cli.py:71]: see here: https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/feedstock_io.py#L53

**Review by beckermr — COMMENTED** [review]
Do we need to delete the conda-forge.yml in the recipe dir?

**Comment by mariusvniekerk** [comment]
> Do we need to delete the conda-forge.yml in the source recipe dir?

Don't think so.  Just removing it from the place where the copy tree copies it into should be fine

File: conda_smithy/cli.py (hunk 2 above)
Comment (mariusvniekerk) [reply conda_smithy/cli.py:73]: Ah gotcha, updated appropriately

File: conda_smithy/cli.py [hunk 3]
```diff
@@ -52,6 +53,25 @@ def generate_feedstock_content(target_directory, source_recipe_dir):
         with feedstock_io.write_file(forge_yml) as fh:
             fh.write("{}")
 
+    # merge in the existing configuration in the source recipe directory
+    forge_yml_recipe = os.path.join(source_recipe_dir, "conda-forge.yml")
+    yaml = YAML()
+    if os.path.exists(forge_yml_recipe):
+        feedstock_io.remove_file(
+            os.path.join(target_recipe_dir, "conda-forge.yml")
+        )
```
Comment (beckermr) [inline conda_smithy/cli.py:62]: You need to read the current one before removing it.

**Review by beckermr — CHANGES_REQUESTED** [review]
I see a bug.

File: conda_smithy/cli.py [hunk 4]
```diff
@@ -52,6 +53,25 @@ def generate_feedstock_content(target_directory, source_recipe_dir):
         with feedstock_io.write_file(forge_yml) as fh:
             fh.write("{}")
 
+    # merge in the existing configuration in the source recipe directory
+    forge_yml_recipe = os.path.join(source_recipe_dir, "conda-forge.yml")
+    yaml = YAML()
+    if os.path.exists(forge_yml_recipe):
+        feedstock_io.remove_file(
+            os.path.join(target_recipe_dir, "conda-forge.yml")
```
Comment (beckermr) [inline conda_smithy/cli.py:61]: This path is `forge_yml` above.

File: conda_smithy/cli.py (hunk 3 above)
Comment (mariusvniekerk) [reply conda_smithy/cli.py:62]: This is the one that is copies needlessly

File: conda_smithy/cli.py (hunk 4 above)
Comment (mariusvniekerk) [reply conda_smithy/cli.py:61]: forge_yml is `target_directory / "conda-forge.yml"`

Comment (beckermr) [reply conda_smithy/cli.py:61]: ack sorry!

**Review by beckermr — APPROVED** [review]

**Review by isuruf — APPROVED** [review]

**Comment by jakirkham** [comment]
Neat! So this would work for merged recipes at staged-recipes?

**Pull request merged**